go-controller/pkg/ovn/ovn.go

```diff
@@ -555,6 +561,56 @@ func (oc *Controller) WatchNetworkPolicy() error {
 	return err
 }
 
+// WatchEgressFirewall starts the watching of egressfirewall resource and calls
+// back the appropriate handler logic
+func (oc *Controller) WatchEgressFirewall() error {
+	_, err := oc.watchFactory.AddEgressFirewallHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			egressFirewall := obj.(*egressfirewall.EgressFirewall)
+			errList := oc.addEgressFirewall(egressFirewall)
+			for _, err := range errList {
+				klog.Error(err)
+				egressFirewall.Status.EgressFirewallObjectStatus = "EgressFirewall not applied correctly, please check logs"
+			}
+			if errList == nil {
+				egressFirewall.Status.EgressFirewallObjectStatus = "EgressFirewall Rules applied"
+			}
+			err := oc.updateEgressFirewallObjectWithRetry(egressFirewall)
+			if err != nil {
+				klog.Error(err)
+			}
+		},
+		UpdateFunc: func(old, newer interface{}) {
+			newEgressFirewall := newer.(*egressfirewall.EgressFirewall)
+			oldEgressFirewall := old.(*egressfirewall.EgressFirewall)
+			if !reflect.DeepEqual(oldEgressFirewall, newEgressFirewall) {
+				errList := oc.updateEgressFirewall(oldEgressFirewall, newEgressFirewall)
+				if errList != nil {
+					newEgressFirewall.Status.EgressFirewallObjectStatus = "EgressFirewall Rules not applied cleanly, please check logs"
+					for _, err := range errList {
+						klog.Error(err)
+					}
+
+				} else {
+					newEgressFirewall.Status.EgressFirewallObjectStatus = "EgressFirewall Rules applied"
```

Comment:
By the way, maybe you could just have a `const` for these strings?

Reply:
done